kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/net/MetadataDiscoveryEndpointBinding.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.net;
+
+import java.util.Objects;
+
+import io.kroxylicious.proxy.service.HostPort;
+
+/**
+ * A bootstrap binding.
+ *
+ * @param endpointGateway the endpoint gateway
+ */
+public record MetadataDiscoveryEndpointBinding(EndpointGateway endpointGateway, Integer nodeId)
```

Comment:
Maybe rename too MetadataDiscoveryBrokerEndpointBinding.  Might be worth linking BrokerEndpointBinding and explaining the role this class has.